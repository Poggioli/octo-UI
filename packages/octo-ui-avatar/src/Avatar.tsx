import { forwardRef } from "@octo-ui/core";
import {
  StyledAvatarFallback,
  StyledAvatarImage,
  StyledAvatarRoot,
} from "./styles";
import { AvatarProps } from "./types";

const Avatar = forwardRef<typeof StyledAvatarRoot, AvatarProps>(
  ({ children, delayMs, css, size, color, ...props }, forwardedRef) => {
    return (
      <StyledAvatarRoot css={css} size={size} ref={forwardedRef}>
        <StyledAvatarImage {...props} />
        <StyledAvatarFallback delayMs={delayMs} color={color}>
          {children}
        </StyledAvatarFallback>
      </StyledAvatarRoot>
    );
  }
);
Avatar.toString = () => `.${StyledAvatarRoot.className}`;

export { StyledAvatarRoot, StyledAvatarImage, StyledAvatarFallback };
export { Avatar };
