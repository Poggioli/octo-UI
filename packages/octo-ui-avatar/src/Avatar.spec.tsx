import { render } from "@testing-library/react";
import { Avatar } from "./Avatar";

describe("<Avatar />", () => {
  it(`GIVEN a <Avatar />
      WHEN render is called
      THEN should render a <Avatar />`, () => {
    const rendered = render(
      <Avatar
        src="https://images.unsplash.com/photo-1535025183041-0991a977e25b?w=300&dpr=2&q=80"
        alt="Landscape photograph by Tobias Tullius"
      >
        NL
      </Avatar>
    );
    expect(rendered.container).toMatchSnapshot();
  });

  it(`GIVEN a <Avatar />
      WHEN render is fallback
      THEN should render a "NL" string`, () => {
    const rendered = render(
      <Avatar
        src="https://imagees.unsplash.com/photo-1535025183041-0991a977e25b?w=300&dpr=2&q=80"
        alt="Landscape photograph by Tobias Tullius"
      >
        NL
      </Avatar>
    );
    expect(rendered.container).toMatchSnapshot();
  });

  it(`GIVEN a <Avatar />
      WHEN toString is called
      THEN return a string`, () => {
    expect(Avatar.toString()).toBe(".octo-UI-c-lmFHv");
  });
});
