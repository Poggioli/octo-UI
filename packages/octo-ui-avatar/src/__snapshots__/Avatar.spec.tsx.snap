// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Avatar /> GIVEN a <Avatar />
      WHEN render is called
      THEN should render a <Avatar /> 1`] = `
<div>
  <span
    class="octo-UI-c-lmFHv octo-UI-c-lmFHv-bvIKlo-size-standard"
  >
    <span
      class="octo-UI-c-caKoxB octo-UI-c-caKoxB-ehJAYl-color-unstyled"
    >
      NL
    </span>
  </span>
</div>
`;

exports[`<Avatar /> GIVEN a <Avatar />
      WHEN render is fallback
      THEN should render a "NL" string 1`] = `
<div>
  <span
    class="octo-UI-c-lmFHv octo-UI-c-lmFHv-bvIKlo-size-standard"
  >
    <span
      class="octo-UI-c-caKoxB octo-UI-c-caKoxB-ehJAYl-color-unstyled"
    >
      NL
    </span>
  </span>
</div>
`;
